File: figures/render-figures.rb
#!/usr/bin/env ruby

require "rexml/document"
include REXML

def rebrand(fname, brand)
  svg = Document.new(File.new(fname, 'r'))
  temp = 
  svg.root.each_element("/svg/g/rect[@id='background']") do |e|
    e.attributes["style"] = "fill:url(##{brand});"
    puts e
  end
  temp_f = File.new(TEMP,'w+')
  temp_f.puts svg
  temp_f.close
  File.delete(TEMP)
end

TEMP = './tmp.svg'

Dir.glob("*svg") do |fname|
  puts fname
  #rebrand(fname, "RHEL7")
  system "inkscape --vacuum-defs -l ../getting-started/C/figures/#{fname} #{fname}"
end

File: figures/render-figures.py
#!/usr/bin/env python3

import glob
import os,shutil
from xml.etree import ElementTree

TEMP = './tmp.svg'

def relativizeSize(fname):
  global svg
  
  #calculate relative sizes if used as xinclude
  root = svg.getroot()
  width = int(root.get('width'))
  height = int(root.get('height'))
  root.set('width','100%')
  root.set('height',str(round( height / width * 100, 2)) + "%")
  root.set('viewBox',"0 0 %s %s" % (width, height))

def rebrand(fname, brand):
  global svg
  
  for e in svg.iterfind(".//{http://www.w3.org/2000/svg}rect[@id='background']"):
    e.set('style', 'fill:url(#%s);' % (brand,))
  

for fname in glob.glob('*svg'):
  global svg
  
  print("processing %s" % (fname))
  os.system("inkscape --vacuum-defs -l %s %s" % (TEMP, fname))
  #plain SVG would strip the itst namespace
  #needed to give context to translators
  #shutil.copyfile(fname,TEMP)
  svg = ElementTree.parse(TEMP)
  #FIXME: thumbs jsou GNOME, zbytek BLANK
  if (fname[:8]=="gs-thumb"):
    rebrand(fname, "GNOME") #BLANK, GNOME, RHEL7
  else:
    rebrand(fname, "BLANK") #BLANK, GNOME, RHEL7

  #relativizeSize(fname)
  svg.write('../gnome-help/C/%s' % (fname))
  os.unlink(TEMP)
